### PR TITLE
Authoring 1936 sidebar references

### PR DESCRIPTION
**Pull request by koz2**
This PR added a list of all locations referencing the currently viewed resource to the document sidebar. 
Currently, clicking any document in the list links the user to it, and clicking any org sets it as the current org. 

Most changes are in ContextAwareSidebar.tsx.

RISK: Low
STABILITY: Medium - it has held up during testing without any unexpected behavior, but has thrown some errors in the console that did not affect the user in any visible manner.  These were mostly from the outstanding lock issue or type errors.

### CODE REVIEW

File: src/components/objectives/ObjectiveSkillView.tsx [hunk 1]
```diff
@@ -61,6 +61,7 @@ const getPoolInfoFromPoolRefEdge = (edge: Edge, questionCount: number): Maybe<Po
   });
 };
 
+// KEVIN-1936 may be needed to filter Edges by type?...
```
Comment (darrensiegel) [inline src/components/objectives/ObjectiveSkillView.tsx]: We should remove these types of comments

File: src/components/objectives/ObjectiveSkillView.tsx [hunk 2]
```diff
@@ -265,7 +266,7 @@ class ObjectiveSkillView
       loading: false,
       organizationResourceMap: Maybe.nothing(),
       skillQuestionRefs: Maybe.nothing(),
-      workbookPageRefs: Maybe.nothing(),
+      workbookPageRefs: Maybe.nothing(), // KEVIN-1936 (BBB) workbookPageRefs initialized as nothing
```
Comment (darrensiegel) [inline src/components/objectives/ObjectiveSkillView.tsx]: Remove

File: src/components/sidebar/ContextAwareSidebar.tsx [hunk 3]
```diff
@@ -33,12 +34,15 @@ import { ToggleSwitch } from 'components/common/ToggleSwitch';
...
+import { Edge } from 'types/edge';
 import { CourseState } from 'reducers/course';
+import { viewDocument, viewOrganizations } from 'actions/view';
+import { getNameAndIconByType } from 'components/ResourceView';
 
 interface SidebarRowProps {
   label?: string;
 }
-
+// KYLE-1936 this is a model usable for the functional component for refs, esp if styles are needed
```
Comment (darrensiegel) [inline src/components/sidebar/ContextAwareSidebar.tsx]: Remove

File: src/components/sidebar/ContextAwareSidebar.tsx [hunk 4]
```diff
@@ -158,12 +164,46 @@ class ContextAwareSidebar
...
+
+      // KEVIN-1936 see (AAA) to see what call to replicate
+      // fetch edges for reference list
+    const fetchResouceEdges = persistence.fetchEdges(course.guid, {
+      // sourceType: LegacyTypes.assessment2,
+      // destinationType: LegacyTypes.assessment2_pool,
+    }).then((edges) => {
+
+      const sources = edges.filter((edge) => {
+        return edge.destinationId.indexOf(resource.id) >= 0;
```
Comment (darrensiegel) [inline src/components/sidebar/ContextAwareSidebar.tsx]: Using 'indexOf' here in this manner will incorrectly identify edges whose destinationIds is the resource.id plus some additional characters.  Rework this to use a string comparison.

File: src/components/sidebar/ContextAwareSidebar.tsx [hunk 5]
```diff
@@ -275,6 +317,88 @@ class ContextAwareSidebar
...
+                            {getRefTitleFromRef(ref)}
+                        </a>
+                      </div>
+                    ))}
+                  </div>
+                ) : (
+                  <div>No references found</div>
+                );
+              },
+              nothing: () => <div>Error loading</div>,
```
Comment (darrensiegel) [inline src/components/sidebar/ContextAwareSidebar.tsx]: I don't think that "Error loading" is appropriate to display here.  resourceRefs is Nothing by default - which means until the network request completes "Error loading" is what will be displayed. You are seeing this in your UI, but it is probably completing so fast that you never notice it.  We should only display "Error loading" when the network request actually has failed.

File: src/components/sidebar/ContextAwareSidebar.tsx (hunk 4 above)
Comment (koz2) [reply src/components/sidebar/ContextAwareSidebar.tsx]: Updated to use string comparison with properly formatted Ids

File: src/components/sidebar/ContextAwareSidebar.tsx (hunk 3 above)
Comment (koz2) [reply src/components/sidebar/ContextAwareSidebar.tsx]: All such tags should be removed now

File: src/components/sidebar/ContextAwareSidebar.tsx (hunk 5 above)
Comment (koz2) [reply src/components/sidebar/ContextAwareSidebar.tsx]: Now cases on new state variable to display error when necessary, loading message otherwise

**Pull request merged**